### PR TITLE
🎨 Make the welcome view adapt to SSO linking

### LINKED ISSUE #648: Make the onboarding (welcome) screen configurable for SSO
## Description
Add a flag in state to tell the onboarding view how to configure itself. Then make the UI respect that flag. Details https://github.com/toggl/mobileapp/issues/7420 and https://github.com/toggl/mobileapp/issues/7421. 

The transition between the two states has to happen with a smooth animation (check the Xamarin app).

## Dependencies
Flag is added to state
View is configured according to the flag (this includes adding the missing pieces, like the cancel button)
Transition between views (after hitting cancel), happens smoothly

### LINKED ISSUE #651: Add CancelButtonTapped action
## Description
Add CancelButtonTapped action https://www.notion.so/toggl/Welcome-63c8191fa8144fd5af36a2f266cad16a?p=330d4694b9794e80b837e714ddeb73b6 
## Dependencies
#648 

## Definition of done
Action is added
It's wired to the UI
It’s reduced so that the onboarding screen is reconfigured to show the regular state
Tests are added

- [ ] The documentation is updated to reflect the changes done in this PR

### PR DESCRIPTION
**Pull request by pdrj**
### Summary
This makes the welcome view adapt to SSO linking

### Technical considerations
It's just using `animateLayoutChanges` to do the animation between states. We can customize it more, but I don't think it's worth it.

### Relationships

Closes #648 
Closes #651

### Pull Request checklist

- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
Bind the configuration state change to a button and observe the layout changes.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Anyone

### CODE REVIEW

**Comment by pdrj** [comment]
After adding tests, also closes #651

File: onboarding/src/main/java/com/toggl/onboarding/welcome/domain/WelcomeReducer.kt [hunk 1]
```diff
@@ -18,5 +18,6 @@ class WelcomeReducer @Inject constructor() : Reducer<WelcomeState, WelcomeAction
                 state.mutateWithoutEffects { copy(backStack = backStack.push(Route.Login)) }
             WelcomeAction.ContinueWithGoogleButtonTapped -> TODO()
             WelcomeAction.LoginWithSsoButtonTapped -> TODO()
```
Comment (daividssilverio) [inline onboarding/src/main/java/com/toggl/onboarding/welcome/domain/WelcomeReducer.kt:20]: 🤔 why not change the configuration already here?

Comment (pdrj) [reply onboarding/src/main/java/com/toggl/onboarding/welcome/domain/WelcomeReducer.kt:20]: You mean on `LoginWithSso` already use the `SsoLink` configuration? That would be wrong, as not all sso logins go through the linking phase. Please let me know if you meant something else.

Comment (pdrj) [reply onboarding/src/main/java/com/toggl/onboarding/welcome/domain/WelcomeReducer.kt:20]: Hm, this PR is concerning the main welcome screen only. It doesn't handle any navigation. The navigation to the sso screen must have happened before in order for user to end up in the welcome screen configured in linking state. Maybe these flows will make it clearer https://miro.com/app/board/o9J_kt0dzyA=/?moveToWidget=3074457348029962191&cot=6

Or maybe I'm not getting your question right 😄

Comment (daividssilverio) [reply onboarding/src/main/java/com/toggl/onboarding/welcome/domain/WelcomeReducer.kt]: Oh, that flow actually helps with my other PR #785. Now I get it.

**Pull request merged**